src/main/java/io/supertokens/webserver/api/core/DeleteUserAPI.java

```diff
@@ -54,8 +54,10 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws I
             UserIdMapping userIdMapping = io.supertokens.useridmapping.UserIdMapping.getUserIdMapping(super.main,
                     userId, UserIdType.ANY);
             if (userIdMapping != null) {
+                // if a userId mapping exists we will need call the deleteUser function with both the superTokensUserId
+                // and the externalId. This will ensure that if the externalId is used in a non-auth recipe, then even
+                // that entry will be deleted.
                 AuthRecipe.deleteUser(main, userIdMapping.superTokensUserId);
-
                 AuthRecipe.deleteUser(main, userIdMapping.externalUserId);
```

Comment:
You should not do this -> instead, you should use the supertokensUserId to delete auth recipe stuff, and external user ID to delete non auth recipe stuff.